src/frontend/screens/Quests/index.tsx

```diff
@@ -117,63 +127,100 @@ export function QuestsPage() {
       )
     }
   })
+  //Search Quest
+  const fuseOptions = {
+    keys: ['name', 'title'],
+    threshold: 0.2
+  }
+
+  const questsWithGameNames =
+    quests?.map((quest) => {
+      const gameName = listings?.[quest.project_id]?.project_meta?.name ?? ''
+      return {
+        ...quest,
+        title: gameName,
+        name: quest.name
+      }
+    }) ?? []
 
-  const gameTitleMatches = (quest: Quest) => {
-    const title = listings ? listings[quest.project_id]?.project_meta?.name : ''
-    return title?.toLowerCase().startsWith(searchText.toLowerCase())
+  const fuse = new Fuse(questsWithGameNames ?? [], fuseOptions)
+  const searchFilteredQuests = searchText
+    ? (fuse.search(searchText) as FuseResult<Quest>[])
+    : quests?.map((quest) => ({
+        id: quest.id,
+        project_id: quest.project_id,
+        name: quest.name,
+        title: listings?.[quest.project_id]?.project_meta?.name ?? '',
+        item: quest,
+        refIndex: 0
+      })) ?? []
+
+  const searchQuests = (quests: Quest[], query: string) => {
+    if (!query) return quests
+    const results = fuse.search(query)
+    return results.map((result) => result.item)
   }
 
-  const searchFilteredQuests = quests?.filter((quest) => {
-    const questTitleMatch = quest.name
-      .toLowerCase()
-      .startsWith(searchText.toLowerCase())
-    const gameTitleMatch = gameTitleMatches(quest)
-    const searchByKeywords = searchText
-      .toLowerCase()
-      .split(' ')
-      .some(
-        (term) =>
-          quest.name?.toLowerCase().includes(term) ||
-          quest.description?.toLowerCase().includes(term)
+  const filteredQuests = searchQuests(quests ?? [], searchText)
+
+  useEffect(() => {
+    if (filteredQuests.length > 0) {
+      const currentQuestExists = filteredQuests.find(
+        (quest) => quest.id === selectedQuestId
       )
-    return questTitleMatch || gameTitleMatch || searchByKeywords
-  })
+      if (!currentQuestExists) {
+        const firstQuest = filteredQuests[0]
+        if (firstQuest) {
+          const searchParams = new URLSearchParams(window.location.search)
+          const newUrl = `/quests/${firstQuest.id}${
+            searchParams.toString() ? `?${searchParams.toString()}` : ''
+          }#card-${firstQuest.id}`
```

Comment:
do we need the `#card-${firstQuest.id}` here?

Reply:
It makes the page scroll to the quest when we search and the page is loaded, this is more used when there is a big list of quests with the same initials for example.